backup-loop.sh

```diff
@@ -207,25 +207,18 @@ tar() {
   backup() {
     ts=$(date +"%Y%m%d-%H%M%S")
     outFile="${DEST_DIR}/${BACKUP_NAME}-${ts}.${backup_extension}"
-    tries=3
-    while (( tries-- > 0 )); do
-      log INFO "Backing up content in ${SRC_DIR} to ${outFile}"
-      command tar "${excludes[@]}" "${tar_parameters[@]}" -cf "${outFile}" -C "${SRC_DIR}" . || exitCode=$?
-      if [ ${exitCode:-0} -eq 0 ]; then
-        break
-      elif [ ${exitCode:-0} -eq 1 ]; then
-        if (( tries > 0 )); then
-          log INFO "...retrying backup in 5 seconds"
-          sleep 5
-          continue
-        else
-          log WARN "Giving up on this round of backup"
-        fi
-      elif [ ${exitCode:-0} -gt 1 ]; then
-        log ERROR "tar exited with code ${exitCode}! Aborting"
-        exit 1
-      fi
-    done
+    log INFO "Backing up content in ${SRC_DIR} to ${outFile}"
+    # sort files so that dat files are archived first
+    (cd "${SRC_DIR}" && { find . -type f -name "*.dat" -o -name "*.dat_old"; find . -type f -not -name "*.dat" -not -name "*.dat_old"; } ) |
+    command tar "${excludes[@]}" "${tar_parameters[@]}" --sort name -cf "${outFile}" -C "${SRC_DIR}" -T - || exitCode=$?
```

Comment:
It seems like `--sort name` will defeat the explicit ordering of the files.

Reply:
Actually, it has no effect when used with `-T`